aleph0/src/prompts.ts

```diff
@@ -0,0 +1,31 @@
+export const createTaskDescriptionPrompt = (options: {
+  userPrompt: string;
+}) => {
+  const { userPrompt } = options;
+
+  return `You are an expert full-stack developer.
+Create a more complete description of this task to pass on to an AI agent.
+The description should be kept to 1-2 lines if possible. Make sure to not deviate from the user's task:
+${userPrompt}
+  
+Assume that the project structure exists already and you are adding files to it or modifying files.
+Task description:`;
+};
+
+export const functionCallPrompt = (options: {
+  userPrompt: string;
+  projectDir: string;
+}) => {
+  const { userPrompt, projectDir } = options;
+
+  return `This is your task: ${userPrompt}
+  
+  * Make sure you use the relevant snippets only for the task. Use the snippets metadata to understand what each snippet does.
+  * Assume that the basic strucure of the project has already been created.
+  * When writing or reading files, use the relative paths.
+  * Only write or read files from this project directory: ${projectDir}
+  * Current working directory: ${process.cwd()}
```

Comment:
We should be enforcing these rules. Not the AI.

We know the folder it should be creating files in, so we should only allow it to create there. We don't need to pass this as a written rule, but as a rule in our own code.

The AI shouldn't be able to create a file somewhere else even if it tried to.

Any AI rules can never be assumed to be 100% followed. So when we can enforce rules of the system ourself, we must do that.

Reply:
I agree, question is how you make sure of that easily? At the end this could be run as a global CLI, so I wanted to give it flexibility to assume and think about the paths itself.

Reply:
@elie222 is there an issue for this?

Reply:
https://github.com/alephmatic/nextjs-ai/issues/40

Reply:
Resolving for now. Elie will fix in another PR